paddlenlp/transformers/convbert/modeling.py

```diff
@@ -0,0 +1,1046 @@
+        mask_positions = paddle.where(generator_labels == -100, umask_positions,
+                                      mask_positions)
+        if mask_positions.sum() == 0:
+            gen_loss = paddle.to_tensor([0.0])
+        else:
+            gen_loss = gen_loss.sum() / mask_positions.sum()
+
+        # discriminator loss
+        seq_length = discriminator_labels.shape[1]
+        disc_loss = self.disc_loss_fct(
+            paddle.reshape(discriminator_prediction_scores, [-1, seq_length]),
+            discriminator_labels.astype(dtype_float), )
+        if attention_mask is not None:
+            umask_positions = paddle.ones_like(discriminator_labels).astype(
+                dtype_float)
+            mask_positions = paddle.zeros_like(discriminator_labels).astype(
+                dtype_float)
+            use_disc_loss = paddle.where(attention_mask, disc_loss,
+                                         mask_positions)
+            umask_positions = paddle.where(attention_mask, umask_positions,
+                                           mask_positions)
+            disc_loss = use_disc_loss.sum() / umask_positions.sum()
+        else:
+            total_positions = paddle.ones_like(discriminator_labels).astype(
+                dtype_float)
+            disc_loss = disc_loss.sum() / total_positions.sum()
+
+        return self.gen_weight * gen_loss + self.disc_weight * disc_loss
+
+
+class ConvBertPooler(Layer):
+    def __init__(self, hidden_size, pool_act="tanh"):
+        super(ConvBertPooler, self).__init__()
+        self.dense = nn.Linear(hidden_size, hidden_size)
+        self.activation = nn.Tanh()
+        self.pool_act = pool_act
+
+    def forward(self, hidden_states):
+        # We "pool" the model by simply taking the hidden state corresponding
+        # to the first token.
+        first_token_tensor = hidden_states[:, 0]
+        pooled_output = self.dense(first_token_tensor)
+        if self.pool_act == "tanh":
+            pooled_output = self.activation(pooled_output)
+        return pooled_output
+
+
+class ConvBertForMultipleChoice(ConvBertPretrainedModel):
+    def __init__(self, convbert, num_choices=2, dropout=None):
+        super(ConvBertPretrainedModel, self).__init__()
+        self.num_choices = num_choices
+        self.convbert = convbert
+        self.pooler = ConvBertPooler(self.convbert.config["hidden_size"])
+        self.dropout = nn.Dropout(dropout if dropout is not None else
+                                  self.convbert.config["hidden_dropout_prob"])
+        self.classifier = nn.Linear(self.convbert.config["hidden_size"], 1)
+        self.init_weights()
+
+    def forward(self,
+                input_ids=None,
+                token_type_ids=None,
+                position_ids=None,
+                attention_mask=None):
+        input_ids = input_ids.reshape(
+            (-1, input_ids.shape[-1]))  # flat_input_ids: [bs*num_choice,seq_l]
+
+        if token_type_ids:
```

Comment:
Please change to 
```python
if token_type_ids is not None:
```
, since PaddlePaddle only supports tensor  which only contains one element when used as the condition of if/while.